commit 02b070ec13363c3bc0f84d9aba9b841e8c821680
Author: John Levon <john.levon@joyent.com>
Date:   2019-07-09T16:57:48+00:00 (3 months ago)
    
    MANTA-2843 Manta zones lacking CPU cap thwart provisioning
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/lib/post-setup/dev-headnode-prov.js b/lib/post-setup/dev-headnode-prov.js
index 22a855e..8a5e5ad 100644
--- a/lib/post-setup/dev-headnode-prov.js
+++ b/lib/post-setup/dev-headnode-prov.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -44,7 +44,8 @@ function makeHeadnodeProvisionable(opts, cb) {
             });
         },
         function tweakCnapiConfig(ctx, next) {
-            if (ctx.cnapi.metadata.ALLOC_FILTER_HEADNODE === false &&
+            if (ctx.cnapi.metadata.ALLOC_FILTER_CAPNESS === false &&
+                ctx.cnapi.metadata.ALLOC_FILTER_HEADNODE === false &&
                 ctx.cnapi.metadata.ALLOC_FILTER_MIN_RESOURCES === false) {
                 progress('CNAPI is already configured to allow headnode ' +
                     'provisioning and over-provisioning');
@@ -57,6 +58,7 @@ function makeHeadnodeProvisionable(opts, cb) {
                 ' and over-provisioning');
             var update = {
                 metadata: {
+                    ALLOC_FILTER_CAPNESS: false,
                     ALLOC_FILTER_HEADNODE: false,
                     ALLOC_FILTER_MIN_RESOURCES: false
                 }
@@ -133,8 +135,8 @@ function do_dev_headnode_prov(subcmd, opts, args, cb) {
 do_dev_headnode_prov.help = (
     'Make the headnode provisionable, for development and testing.\n' +
     '\n' +
-    'This is done via `ALLOC_FILTER_HEADNODE` and \n' +
-    '`ALLOC_FILTER_MIN_RESOURCES` SAPI configuration of the CNAPI service.\n' +
+    'This is done via `ALLOC_FILTER_CAPNESS`, `ALLOC_FILTER_HEADNODE`, and \n' +
+    '`ALLOC_FILTER_MIN_RESOURCES` SAPI parameters of the CNAPI service.\n' +
     'See https://github.com/joyent/sdc-cnapi/blob/master/docs/index.md#sapi-configuration\n' +
     '\n' +
     'Usage:\n' +
diff --git a/man/man1/sdcadm.1.ronn b/man/man1/sdcadm.1.ronn
index f353931..372e7ae 100644
--- a/man/man1/sdcadm.1.ronn
+++ b/man/man1/sdcadm.1.ronn
@@ -573,10 +573,11 @@ Create the "cns" service and a first instance.
 
 ### sdcadm post-setup dev-headnode-prov
 
-Make the headnode provisionable, for development and testing.
+Make the headnode provisionable, for development and testing. This allows
+a Manta deployment to co-exist with standard Triton provisioning.
 
-This is done via `ALLOC_FILTER_HEADNODE` and `ALLOC_FILTER_MIN_RESOURCES`
-SAPI configuration of the CNAPI service. See
+This is done via `ALLOC_FILTER_CAPNESS`, `ALLOC_FILTER_HEADNODE`,
+and `ALLOC_FILTER_MIN_RESOURCES` SAPI parameters of the CNAPI service. See
 [SAPI configuration](https://github.com/joyent/sdc-cnapi/blob/master/docs/index.md#sapi-configuration )
 
 ### sdcadm platform \[options\] COMMAND \[args...\]
@@ -1135,4 +1136,4 @@ Enables/disables support for various NFS volumes features.
 
 ## COPYRIGHT
 
-sdcadm Copyright (c) 2018 Joyent Inc., All rights reserved.
+sdcadm Copyright 2019 Joyent, Inc.
diff --git a/test/post-setup.test.js b/test/post-setup.test.js
index 80dbba9..ff4b2b8 100644
--- a/test/post-setup.test.js
+++ b/test/post-setup.test.js
@@ -183,7 +183,8 @@ test('sdcadm post-setup dev-headnode-prov', function (t) {
 
             var svc = JSON.parse(stdout)[0];
 
-            if (svc.metadata.ALLOC_FILTER_HEADNODE === false &&
+            if (svc.metadata.ALLOC_FILTER_CAPNESS === false &&
+                svc.metadata.ALLOC_FILTER_HEADNODE === false &&
                 svc.metadata.ALLOC_FILTER_MIN_RESOURCES === false) {
                 t.end();
                 return;
